src/Fixer/Operator/OperatorLinebreakFixer.php

```diff
@@ -0,0 +1,314 @@
+<?php
+
+/*
+ * This file is part of PHP CS Fixer.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *     Dariusz Rumiński <dariusz.ruminski@gmail.com>
+ *
+ * This source file is subject to the MIT license that is bundled
+ * with this source code in the file LICENSE.
+ */
+
+namespace PhpCsFixer\Fixer\Operator;
+
+use PhpCsFixer\AbstractFixer;
+use PhpCsFixer\Fixer\ConfigurationDefinitionFixerInterface;
+use PhpCsFixer\FixerConfiguration\FixerConfigurationResolver;
+use PhpCsFixer\FixerConfiguration\FixerOptionBuilder;
+use PhpCsFixer\FixerDefinition\CodeSample;
+use PhpCsFixer\FixerDefinition\FixerDefinition;
+use PhpCsFixer\Preg;
+use PhpCsFixer\Tokenizer\Analyzer\Analysis\CaseAnalysis;
+use PhpCsFixer\Tokenizer\Analyzer\ReferenceAnalyzer;
+use PhpCsFixer\Tokenizer\Analyzer\SwitchAnalyzer;
+use PhpCsFixer\Tokenizer\Token;
+use PhpCsFixer\Tokenizer\Tokens;
+
+/**
+ * @author Kuba Werłos <werlos@gmail.com>
+ */
+final class OperatorLinebreakFixer extends AbstractFixer implements ConfigurationDefinitionFixerInterface
+{
+    /**
+     * @internal
+     */
+    const BOOLEAN_OPERATORS = [[T_BOOLEAN_AND], [T_BOOLEAN_OR], [T_LOGICAL_AND], [T_LOGICAL_OR], [T_LOGICAL_XOR]];
+
+    /**
+     * @internal
+     */
+    const NON_BOOLEAN_OPERATORS = ['%', '&', '*', '+', '-', '.', '/', ':', '<', '=', '>', '?', '^', '|', [T_AND_EQUAL], [T_CONCAT_EQUAL], [T_DIV_EQUAL], [T_DOUBLE_ARROW], [T_IS_EQUAL], [T_IS_GREATER_OR_EQUAL], [T_IS_IDENTICAL], [T_IS_NOT_EQUAL], [T_IS_NOT_IDENTICAL], [T_IS_SMALLER_OR_EQUAL], [T_MINUS_EQUAL], [T_MOD_EQUAL], [T_MUL_EQUAL], [T_OBJECT_OPERATOR], [T_OR_EQUAL], [T_PAAMAYIM_NEKUDOTAYIM], [T_PLUS_EQUAL], [T_POW], [T_POW_EQUAL], [T_SL], [T_SL_EQUAL], [T_SR], [T_SR_EQUAL], [T_XOR_EQUAL]];
```

Comment:
is this complete list, what about `**`, `instanceof`, `<=>`?

Reply:
what about them? do you want `**` and `<=>` to be removed and `instanceof` to be added? ;)

That does not make a sense - `instanceof` is not operator according to [docs](https://www.php.net/manual/en/language.operators.php) - it would make sense to add it as it works like operator or wouldn't? Ping @SpacePossum and @julienfalque for an opinion.

Reply:
I see, spaceship is on line 107 and `**` is probably covered by `T_POW`. As this list contains also operators like `::` (`T_PAAMAYIM_NEKUDOTAYIM`), then I think `instanceof` should be added.

at least for non-boolean operators, I would very like to see a support for removing the line break completely.

Reply:
Once again `::` is on the list of operators and `instanceof` - I think we can extend the list, but let's merge this with the list we have (all the operators from https://www.php.net/manual/en/language.operators.php) and then discuss new items separately.

> Also, at least for non-boolean operators, I would very like to see a support for removing the line break completely.

Good idea, with the `position` value `none` and similarly like above, let's do it in separate PR after this one is merged.

Reply:
> Good idea, with the `position` value `none` and similarly like above, let's do it in separate PR after this one is merged.

I support merging this asap, but I think we would need a different config to support different `position` for different types. Not sure how much breaking changes are an issue in `master`, but I think the config should change to something like:
- `position_boolean_operator` =>`beginning`
- `position_other_operator` =>`none` (or maybe `remove`, also, this must support false)

with these values beiing default and drop `only_booleans` completely.

Reply:
Breaking changes are only for released stuff only - if this is merged we can change it as much as we can until it is released.

Reply:
Nice suggestion!
I think leaving out `instance of` is a good call, but happy to see real life examples where this could add value.
For the new options, we can adjust until released and even after that we can add new options and deprecate the current ones (less ideal surely). As such, maybe best to start a new issue/PR for expected changes and discuss there, thanks!